src/main/java/org/broadinstitute/hellbender/utils/test/SamAssertionUtils.java

```diff
@@ -135,4 +109,126 @@ public static void assertSamValid(final File sam) throws IOException {
+                if (eqMessage != null){
+                    return eqMessage;
+                }
+            }
+            if (it1.hasNext() || it2.hasNext()) {
+                //at least one has no more records (because the while loop is done) and at least one does have records. So we're not equal.
+                return "Not the same number of reads";
+            }
+            return null;
+        }
+    }
+
+    /**
+     * Compares the reads but ignores order or attributes.
+     * Also allows read2 to have a superset of attributes of read1.
+     */
+    private static String readsEqualAllowAddingAttributes(final SAMRecord read1, final SAMRecord read2) {
+        final SAMRecordCoordinateComparator coordinateComparator = new SAMRecordCoordinateComparator();
+        if (!Objects.equals(read1.getReadName(), read2.getReadName())){
+            return "name different read1:" + read1.getReadName() + " read2:" + read2.getReadName() ;
+        }
+        if (!Objects.equals(read1.getFlags(), read2.getFlags())){
+            return "flags different read1:" + read1.getReadName() + " read2:" + read2.getReadName() ;
+        }
+        if (!Objects.equals(read1.getInferredInsertSize(), read2.getInferredInsertSize())){
+            return "getInferredInsertSize different read1:" + read1.getReadName() + " read2:" + read2.getReadName() + " (" + read1.getInferredInsertSize()+ " vs " + read2.getInferredInsertSize() + ")";
+        }
+        if (!Objects.equals(read1.getMappingQuality(), read2.getMappingQuality())){
+            return "getMappingQuality different read1:" + read1.getReadName() + " read2:" + read2.getReadName() ;
+        }
+        //Note the comparator does check flags, MQ and insert size but we want better messages
+        if (0 != coordinateComparator.fileOrderCompare(read1, read2)){
+            return "file order read1:" + read1.getContig() + ":" + read1.getAlignmentStart() + " read2:" + read2.getContig() + ":" + read2.getAlignmentStart();
+        }
+        if (!Objects.equals(read1.getCigar(), read2.getCigar())){
+            return "getCigar different read1:" + read1.getReadName() + " read2:" + read2.getReadName() ;
+        }
+        if (!Arrays.equals(read1.getReadBases(), read2.getReadBases())){
+            return "getReadBases different read1:" + read1.getReadName() + " read2:" + read2.getReadName() ;
+        }
+        if (!Arrays.equals(read1.getBaseQualities(), read2.getBaseQualities())){
+            return "getBaseQualities different read1:" + read1.getReadName() + " read2:" + read2.getReadName() ;
+        }
+        final List<SAMRecord.SAMTagAndValue> attr1 = read1.getAttributes();
+        final List<SAMRecord.SAMTagAndValue> attr2 = read2.getAttributes();
+        if (attr1.size() > attr2.size()){
+            return "lost attributes read1:" + read1.getReadName() + " read2:" + read2.getReadName() + "("+ attr1.size() + " vs " + attr2.size() +")" ;
```

Comment:
This message could be worded more clearly -- "read1 contains attributes that read2 lacks"